engine/src/state_chain_observer/client.rs

```diff
@@ -276,55 +297,125 @@ impl<BaseRpcClient: base_rpc_api::BaseRpcApi + Send + Sync + 'static, SignedExtr
 
 		let mut processed_stream = {
 			let latest_block: BlockInfo = block_stream.next().await.unwrap()?;
-			let mut block_stream =
-				process_stream_fn(Box::pin(block_stream.make_try_cached(latest_block))).await?;
-
-			let latest_block = *block_stream.cache();
-			if wait_for_required_version {
-				let incompatible_blocks =
-					futures::stream::once(futures::future::ready(Ok::<_, anyhow::Error>(
-						latest_block,
-					)))
-					.chain(block_stream.by_ref())
-					.try_take_while(|block_info| {
-						let base_rpc_client = base_rpc_client.clone();
-						let block_info = *block_info;
-						async move {
-							Ok(
-								if let Err(block_version) = base_rpc_client
-									.check_block_compatibility(block_info.hash)
+
+			let block_stream: Pin<
+				Box<
+					dyn TryCachedStream<
+							Error = anyhow::Error,
+							Ok = BlockInfo,
+							Item = Result<BlockInfo, anyhow::Error>,
+						> + Send,
+				>,
+			> = if let Some(start_from) = start_from {
+				let base_rpc_client = base_rpc_client.clone();
+
+				let mut start_from_headers = Box::pin(
```

Comment:
Because we know of node caching problems, I feel we probably want to not fail if the node has deleted some of the blocks between start_from and latest_block, because we could get into a bad situation where engines cannot start.

Reply:
Or atleast we need some way to allow the engine / single binary to try without the start_from if it fails.

Reply:
This will only be the case if there is a `start_from`. Basically the start_from would only ever be provided on the handover. So if the engine exited and was restarted it'd be started directly from the new version, and therefore not have a `start_from` and so it would just start as it does now.